metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py

```diff
@@ -127,16 +121,17 @@ def generate_partition_profiler_query(
                     return None, None
 
                 # ingestion time partitoned tables partition column is not in the schema, so we default to TIMESTAMP type
-                partition_column_type: str = "TIMESTAMP"
-                for c in table.columns:
-                    if c.is_partition_column:
-                        partition_column_type = c.data_type
+                if not table.time_partitioning.column:
+                    logger.warning(
+                        f"Partitioned table {table.name} without partition column, it seems like a bug in our extraction"
+                    )
+                    return None, None
 
-                if table.time_partitioning.type_ in ("HOUR", "DAY", "MONTH", "YEAR"):
-                    partition_where_clause = f"{partition_column_type}(`{table.time_partitioning.field}`) BETWEEN {partition_column_type}('{partition_datetime}') AND {partition_column_type}('{upper_bound_partition_datetime}')"
+                if table.time_partitioning.type in ("HOUR", "DAY", "MONTH", "YEAR"):
```

Comment:
Please ensure we have sufficient try: except around this method. There are a LOT of new changes and it makes me quite nervous.

Reply:
I'm wouldn't worry about this now as here we have typed objects with non-optional properties, and earlier, we used Bigquery's object, which was backed by a dict. (that is where the `key` error happened because it's to string method failed)